Dockerfile

```diff
@@ -124,4 +124,4 @@ FROM scratch AS plugins
 COPY --from=build-plugins /out .
 
 FROM scratch AS binary
-COPY --from=build /out .
+COPY --from=build /out/docker /docker
```

Comment:
Curious why this `COPY` had to be changed; did we add more things in `/out` ? Where possible,  I like “copy all the things” so that the part copying things doesn’t need to know what to look for

(basically "`/out`" being the "contract" / API (for better words)

Reply:
`scripts/build/binary`  produces two files, first is `docker-<platform>-<arch>` which is an actual binary, and the second is a `docker` which is a symlink to the former.

So without this change the image content would be:
```
/docker-<platform>-<arch>        # the actual binary
/docker                          # symlink to the above
```


The `COPY` copies the /docker which dereferences into the actual binary and puts that directly as `/docker` in the final image. So the image only contains the `docker` binary directly.

It's not strictly needed, because it should work with `COPY --from=dockereng/cli /docker /docker` anyway, but I'm not sure if we really need that symlink in the final image.


Alternatively I could look into changing the `scripts/build/binary` so that it doesn't output the symlink, but that would probably require changing other things.

Reply:
Ah, yes, forgot about the symlink; yeah. not a blocker, just something "nice if we can improve"

Reply:
LOL, and of course I didn't think of docker-ce-packaging using this output, and now failng;

```
dest=$PWD/build/mac; cd /home/ubuntu/workspace/release-packaging_ce-nightly/packaging/src/github.com/docker/cli/build && for platform in *; do \
	arch=$(echo $platform | cut -d_ -f2); \
	mkdir -p $dest/$arch/docker; \
	cp $platform/docker-darwin-* $dest/$arch/docker/docker && \
	tar -C $dest/$arch -c -z -f $dest/$arch/docker-25.0.0-rc.2.tgz docker; \
done
cp: cannot stat 'darwin_amd64/docker-darwin-*': No such file or directory
cp: cannot stat 'darwin_arm64/docker-darwin-*': No such file or directory
```